crates/aranya-daemon/benches/metrics.rs

```diff
@@ -0,0 +1,42 @@
+use std::net::Ipv4Addr;
+
+use anyhow::Result;
+use aranya_daemon::{
+    config::{AqcConfig, Config, QuicSyncConfig, SyncConfig, Toggle},
+    Daemon,
+};
+use aranya_util::Addr;
+use divan::AllocProfiler;
+use tokio::runtime::Runtime;
+
+#[global_allocator]
+static ALLOC: AllocProfiler = AllocProfiler::system();
+
+fn main() {
+    divan::main();
+}
+
+// NOTE: divan currently requires sync functions to work, so we spawn a runtime and block on async.
+#[divan::bench]
+fn daemon_startup() -> Result<()> {
+    let work_dir = tempfile::tempdir()?.path().to_path_buf();
+
+    let cfg = Config {
+        name: "daemon".into(),
+        runtime_dir: work_dir.join("run"),
+        state_dir: work_dir.join("state"),
+        cache_dir: work_dir.join("cache"),
+        logs_dir: work_dir.join("log"),
+        config_dir: work_dir.join("config"),
+        sync: SyncConfig {
+            quic: Toggle::Enabled(QuicSyncConfig {
+                addr: Addr::from((Ipv4Addr::UNSPECIFIED, 0)),
+            }),
+        },
+        aqc: Toggle::Enabled(AqcConfig {}),
+    };
+
+    let rt = Runtime::new()?;
+    let _daemon = rt.block_on(Daemon::load(cfg))?;
```

Comment:
This is loading a daemon object from a config file but not starting up the daemon.
```suggestion
    let _daemon = rt.block_on(Daemon::load(cfg).await?.spawn())?;
```

Reply:
@nikki-aranya I thought this feedback was included but it appears to have been reverted at some point. `Daemon::load(cfg)` loads a daemon from a config. `spawn()` needs to be called to run the daemon.

Reply:
Startup suggests that this is measuring how long it takes for the daemon to startup. The following is not currently included in the measurement:
1. How long it takes to initialize the daemon IPC server
2. How long it takes to start up the QUIC sync server

Measuring how long it takes to load the daemon's config file is not a very useful measurement.